frontend/src/app/MindMap.js

```diff
@@ -22,9 +42,15 @@ class MindMap extends React.Component {
             loading: false,
             showNewSourceForm: false,
             showNewSourceHelperMessage: false,
-            newSourceData: null
-        }
-    }
+            newSourceData: modes.NULL,
+            modes:modes,
+            newSourceFormType: null,
+            source: null,
```

Comment:
I can't really tell what this state variable is supposed to be. Maybe give it a more descriptive name.

Reply:
which variable are you referring to?